### PR TITLE
Upgrade TypeScript and ESLint config

### PR DESCRIPTION
**Pull request by mrm007**
Most changes are the result of `eslint --fix`, with a few highlighted exceptions.

### CODE REVIEW

File: site/src/App/routes/guides/playroom-prototyping.tsx [hunk 1]
```diff
@@ -215,7 +215,7 @@ const PlayroomPrototyping = () => (
         return {
           ...result,
           code: result.code
-            .replace(/ as Array<MockJob>/g, '')
+            .replace(/ as MockJob\[\]/g, '')
```
Comment (mrm007) [inline site/src/App/routes/guides/playroom-prototyping.tsx:218]: This will be handled by the `source.macro` package (in a separate PR)

File: package.json [hunk 2]
```diff
@@ -76,13 +76,17 @@
...
   },
   "volta": {
     "node": "18.15.0"
   },
   "packageManager": "pnpm@7.30.0",
   "pnpm": {
+    "overrides": {
+      "@braid-design-system/generate-component-docs": "workspace:*",
+      "braid-design-system": "workspace:*"
+    },
```
Comment (mrm007) [inline package.json:89]: Had to do these shenanigans until `manypkg` supports the `workspace:` protocol https://github.com/Thinkmill/manypkg/pull/186

File: packages/generate-component-docs/src/__snapshots__/contract.test.ts.snap [hunk 3]
```diff
@@ -8207,12 +8209,12 @@ exports[`filterSuggestions 1`] = `
   props: {
```
Comment (mrm007) [inline packages/generate-component-docs/src/__snapshots__/contract.test.ts.snap:8209]: Looks like this export is incorrectly treated as a component, hence the bogus definition for "props".

File: packages/braid-design-system/package.json [hunk 4]
```diff
@@ -200,7 +200,7 @@
     "fs-extra": "^10.1.0",
     "html-validate": "^7.1.1",
     "playroom": "0.32.1",
-    "prettier": "^2.8.2",
+    "prettier": "^2.8.8",
```
Comment (mrm007) [inline packages/braid-design-system/package.json:203]: GitHub wouldn't let me merge because it thought `pnpm-lock.yaml` had conflicts (it didn't) so I made an unrelated change to update the lockfile.

**Review by michaeltaranto — APPROVED** [review]
👏
